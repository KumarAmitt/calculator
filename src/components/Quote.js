import React from 'react';

const Quote = () => {
  const quote = 'Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.';
  const cite = 'William Paul Thurston';
  return (
    <>
      <blockquote data-testid="quote">
        {quote}
        <cite>
          &mdash;
          {cite}
        </cite>
      </blockquote>
    </>
  );
};

export default Quote;
