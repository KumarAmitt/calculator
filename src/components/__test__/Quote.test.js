import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Quote from '../Quote';

test('look for a quote element', () => {
  const { getByTestId } = render(<Quote />);
  const quoteElement = getByTestId('quote');

  expect(quoteElement).toBeInTheDocument();
});

test('matches snapshot of Quote', () => {
  const tree = renderer.create(<Quote />).toJSON();

  expect(tree).toMatchSnapshot();
});
