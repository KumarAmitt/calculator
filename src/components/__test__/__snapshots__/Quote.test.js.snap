// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of Quote 1`] = `
<blockquote
  data-testid="quote"
>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding.
  <cite>
    â€”
    William Paul Thurston
  </cite>
</blockquote>
`;
